tests/run/partial_circular_import.srctree

```diff
@@ -0,0 +1,24 @@
+PYTHON -c "import pkg.A"
```

Comment:
It looks like this requires a somewhat recent version of Python for the Python tests to work. (It's not a concern for the Cython side of it)

```suggestion
PYTHON -c "import sys; if sys.version_info >= (3,5): import pkg.A"
```

Reply:
Looks like the presence of pkg/__init__.py causes the problem for some reason (but it didn't seem to be a problem in the first round of commits).